VideoWeb/VideoWeb/ClientApp/src/app/shared/header/header.component.spec.ts

```diff
@@ -36,3 +42,42 @@ describe('HeaderComponent', () => {
         expect(router.navigate).toHaveBeenCalledWith([component.topMenuItems[0].url]);
     });
 });
+
+describe('Header component template file', () => {
+    let fixture: ComponentFixture<HeaderComponent>;
+    let component: HeaderComponent;
+    let debugElement: DebugElement;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            imports: [RouterTestingModule],
+            declarations: [HeaderComponent, TranslatePipeMock]
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(HeaderComponent);
+        component = fixture.componentInstance;
+        debugElement = fixture.debugElement;
+    });
+
+    it('logout button should route to logout page', () => {
+        component.loggedIn = true;
+
+        fixture.detectChanges();
+
+        const logoutButton = debugElement.query(By.css('#logout_link'));
+
+        expect(logoutButton.nativeElement.pathname).toBe(`/${pageUrls.Logout}`);
+    });
+
+    it('hides logout button when logged in is false', () => {
+        component.loggedIn = false;
```

Comment:
intentionally setting this value explicitly as I think it make it clear that this setup is required for this unit test